src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -430,41 +439,96 @@ func (c controller) sendCompleteInstallation(isSuccess bool, errorInfo string) {
 	c.log.Infof("Done complete installation step")
 }
 
-// get pod logs,
-// write tar.gz to pipe in a routine
-// upload tar.gz from pipe to assisted service.
-// close read and write pipes
-func (c controller) uploadPodLogs(podName string, namespace string, sinceSeconds int64) error {
+func (c controller) uploadSummaryLogs(podName string, namespace string, sinceSeconds int64, isMustGatherEnabled bool) error {
+	var errz = make([]error, 0)
+	var tarentries = make([]utils.TarEntry, 0)
+	ctx := utils.GenerateRequestContext()
+
+	var returnVal = func() error {
+		//report combined errors
+		var cerr error
+		for _, e := range errz {
+			cerr = errors.Errorf("%#v", e)
+		}
+		return cerr
+	}
+
 	c.log.Infof("Uploading logs for %s in %s", podName, namespace)
-	podLogs, err := c.kc.GetPodLogsAsBuffer(namespace, podName, sinceSeconds)
-	if err != nil {
-		return errors.Wrapf(err, "Failed to get logs of pod %s", podName)
+	if podLogs, err := c.kc.GetPodLogsAsBuffer(namespace, podName, sinceSeconds); err == nil {
+		tarentries = append(tarentries,
+			*utils.NewTarEntry(podLogs, nil, int64(podLogs.Len()), fmt.Sprintf("%s.logs", podName)))
+	} else {
+		errz = append(errz, errors.Wrapf(err, "Failed to get logs of pod %s", podName))
+	}
+
+	if isMustGatherEnabled {
+		if tarfile, err := c.collectMustGatherLogs(ctx); err == nil {
+			if entry, tarerr := utils.NewTarEntryFromFile(tarfile); tarerr == nil {
+				tarentries = append(tarentries, *entry)
+			} else {
+				errz = append(errz, errors.Wrapf(tarerr, "Failed to open must-gather logs tar"))
+			}
+		} else {
```

Comment:
u print log for error inside collectMustGatherLogs, not sure why need one more log line? lets have a little less if's here

Reply:
it help us to see where we are in the process... I can remove it but I recommend to keep it for debugging purposes

Reply:
but we have it already inside collectMustGatherLogs ? why to have 2 same logs?